10/07/2025

# 📊 Projeto: Análise de Desempenho da Loja "TecnoFácil"

## Contexto

A TecnoFácil é uma loja de pequeno porte que vende eletrônicos e acessórios. 

A gerência quer entender melhor os dados de vendas, clientes e desempenho ao longo dos meses. Você foi contratado como analista de dados para transformar dados brutos em informações úteis.

Essa ideia de projeto foi gerado pelo `ChatGPT`, para que eu pudesse colocar em prática todo o meu aprendizado feito nos cursos sobre Pandas, NumPy e Matplotlib.

## 🗂️ Dados do Projeto

In [1]:
import pandas as pd

In [2]:
base = pd.read_csv("dados_vendas.csv")

In [3]:
base.head()

Data da Venda           Produto    Categoria  Quantidade  Preço Unitário  \
0    2024-09-07       Notebook X1  Informática           1          3200.0   
1    2025-06-08  Teclado Mecânico   Acessórios           2           350.0   
2    2025-06-21       Notebook X1  Informática           5          3200.0   
3    2024-12-24         Tablet M7    Celulares           4          1450.0   
4    2025-02-20    Fone Bluetooth   Acessórios           1           150.0   

   Receita    Custo             Cliente Estado  
0   3200.0   2700.0        Bruno Mendes     BA  
1    700.0    500.0      Emilly Freitas     MG  
2  16000.0  13500.0    Isabella Almeida     RJ  
3   5800.0   4400.0          Yago Cunha     SP  
4    150.0     90.0  Emanuelly Teixeira     RS

In [20]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Data da Venda   100 non-null    object 
 1   Produto         100 non-null    object 
 2   Categoria       100 non-null    object 
 3   Quantidade      100 non-null    int64  
 4   Preço Unitário  100 non-null    float64
 5   Receita         100 non-null    float64
 6   Custo           100 non-null    float64
 7   Cliente         100 non-null    object 
 8   Estado          100 non-null    object 
dtypes: float64(3), int64(1), object(5)
memory usage: 7.2+ KB


In [26]:
# Quais são as colunas da nossa base de dados:
base.columns

Index(['Data da Venda', 'Produto', 'Categoria', 'Quantidade', 'Preço Unitário',
       'Receita', 'Custo', 'Cliente', 'Estado'],
      dtype='object')

In [15]:
# Apenas testando algo do pandas

base.groupby('Produto')[["Quantidade", "Custo"]].sum()

Quantidade     Custo
Produto                               
Cabo USB-C                34     680.0
Caixa de Som              29    4350.0
Fone Bluetooth            30    2700.0
Impressora Wi-Fi          21   12600.0
Monitor Full HD           21   14700.0
Mouse Gamer               35    2450.0
Notebook X1               41  110700.0
Smartphone Z5             14   25200.0
Tablet M7                 28   30800.0
Teclado Mecânico          48   12000.0

In [23]:
# Quantidade de produtos por categoria

base[["Categoria"]].value_counts()

Categoria  
Acessórios     60
Informática    25
Celulares      15
Name: count, dtype: int64

Com isso podemos visualizar a quantidade de produtos por categoria, e a categoria acessórios é a que tem mais itens (60% do total)

In [22]:
# Verificando se existem valores nulos (NaN)
base.isnull().sum() # Contar os valores ausentes

Data da Venda     0
Produto           0
Categoria         0
Quantidade        0
Preço Unitário    0
Receita           0
Custo             0
Cliente           0
Estado            0
dtype: int64

Com isso, posso observar que não existem nenhum valor NaN (nulo), então já é um caminho bom.

In [34]:
# Fazendo um agrupamento (por Produto e Categoria) e me retornando a soma de todas as linhas de 'Quantidade', 'Custo' e 'Receita'. 
# Além de está sendo ordenado do menor para o maior pelo 'Custo' que a empresa tem por cada produto.

base.groupby(['Produto', 'Categoria'])[['Quantidade', 'Custo', 'Receita']].sum().sort_values(by=['Quantidade'])

Quantidade     Custo   Receita
Produto          Categoria                                  
Smartphone Z5    Celulares            14   25200.0   29400.0
Impressora Wi-Fi Informática          21   12600.0   16380.0
Monitor Full HD  Informática          21   14700.0   19950.0
Tablet M7        Celulares            28   30800.0   40600.0
Caixa de Som     Acessórios           29    4350.0    6090.0
Fone Bluetooth   Acessórios           30    2700.0    4500.0
Cabo USB-C       Acessórios           34     680.0    1530.0
Mouse Gamer      Acessórios           35    2450.0    4200.0
Notebook X1      Informática          41  110700.0  131200.0
Teclado Mecânico Acessórios           48   12000.0   16800.0

### Vamos a uma rápida análise de vendas

Uma **OBS** a detalhar: O Custo, como o nome sugere, é quanto vale o produto para a empresa adquirir. A receita é quanto eu ganhei ao vender para o cliente (Quantidade * Preço Unitário).

- Dentre os produtos, o que mais tem, em **quantidade vendidas** são o seguinte:
    1. Teclado Mecânico	- Acessórios - 48 unidades;
    2. Notebook X1 - Informática - 41;
    3. Mouse Gamer - Acessórios - 35.


- Outra coisa que observei é que, os clientes sempre vão atrás de `Acessórios`, gastando mais um pouco nisso.

In [37]:
base.groupby(['Cliente', 'Estado'])[['Produto']].value_counts()

Cliente             Estado  Produto         
Amanda Oliveira     BA      Notebook X1         1
Ana Clara Lopes     RJ      Cabo USB-C          1
Ana Clara Santos    BA      Caixa de Som        1
Ana Julia Ferreira  RJ      Teclado Mecânico    1
Ana Julia Mendes    CE      Mouse Gamer         1
                                               ..
Vicente Duarte      DF      Monitor Full HD     1
Vinicius Carvalho   SC      Teclado Mecânico    1
Vitor Rocha         RJ      Teclado Mecânico    1
Vitória Santos      RJ      Impressora Wi-Fi    1
Yago Cunha          SP      Tablet M7           1
Name: count, Length: 100, dtype: int64

Dentro dessa base de dados Ficticias, não se tem clientes repetidos.

___

## 🎯 Objetivos

### 📌 Parte 1: Preparação dos Dados (Pandas + NumPy)

- Ler os dados com pd.read_csv()
- Converter a coluna de data com pd.to_datetime()
- Calcular colunas novas:
    - Lucro = Receita - Custo
    - Mês da venda
- Usar NumPy para:
    - Calcular média, desvio padrão e valores únicos de categorias
    - Criar array de lucro e fazer análises manuais (média, max, min)

In [38]:
# Converter a coluna de data:
base['Data da Venda'] = pd.to_datetime(base['Data da Venda'])

In [39]:
# VErificando o tipo das colunas
base.dtypes

Data da Venda     datetime64[ns]
Produto                   object
Categoria                 object
Quantidade                 int64
Preço Unitário           float64
Receita                  float64
Custo                    float64
Cliente                   object
Estado                    object
dtype: object

In [41]:
# Calcular novas colunas: LUCRO

base['Lucro'] = base['Receita'] - base['Custo']

In [42]:
base.head()

Data da Venda           Produto    Categoria  Quantidade  Preço Unitário  \
0    2024-09-07       Notebook X1  Informática           1          3200.0   
1    2025-06-08  Teclado Mecânico   Acessórios           2           350.0   
2    2025-06-21       Notebook X1  Informática           5          3200.0   
3    2024-12-24         Tablet M7    Celulares           4          1450.0   
4    2025-02-20    Fone Bluetooth   Acessórios           1           150.0   

   Receita    Custo             Cliente Estado   Lucro  
0   3200.0   2700.0        Bruno Mendes     BA   500.0  
1    700.0    500.0      Emilly Freitas     MG   200.0  
2  16000.0  13500.0    Isabella Almeida     RJ  2500.0  
3   5800.0   4400.0          Yago Cunha     SP  1400.0  
4    150.0     90.0  Emanuelly Teixeira     RS    60.0

In [55]:
# Calcular novas colunas: Mês da Venda
base.groupby([base['Data da Venda'].dt.month, 'Categoria'])[['Receita']].sum()

Receita
Data da Venda Categoria           
1             Acessórios    3290.0
              Celulares    16450.0
2             Acessórios    2600.0
              Informática  16000.0
3             Acessórios    2280.0
              Informática   3800.0
4             Acessórios    3820.0
              Informática  12100.0
5             Acessórios    2495.0
              Celulares     9200.0
              Informática  29750.0
6             Acessórios    2780.0
              Informática  49560.0
7             Acessórios    2470.0
              Informática   4750.0
8             Acessórios    1305.0
              Celulares     7250.0
              Informática   3120.0
9             Acessórios    2645.0
              Celulares     5000.0
              Informática   9440.0
10            Acessórios    2565.0
              Informática  20750.0
11            Acessórios    5940.0
              Celulares    20000.0
              Informática  14360.0
12            Acessórios     930.0
              Celulares    12100.0
              Informática   3900.0

___

### 📌 Parte 2: Análises (Pandas)

- Qual foi o produto mais vendido?
- Qual foi o mês com mais vendas?
- Qual foi a categoria mais lucrativa?
- Qual cliente mais comprou?
- Qual estado gerou mais receita?

In [61]:
# Qual foi o produto mais vendido?
produtos_mais_vendidos = base.groupby(['Produto'])[['Quantidade', 'Receita']].sum().sort_values(by=['Quantidade'])

In [83]:
produtos_mais_vendidos.tail(1)

Quantidade  Receita
Produto                              
Teclado Mecânico          48  16800.0

O produto que mais foi vendido para os clientes foi o _Teclado Mecânico_, gerando uma receita de R$ 16.800,00

In [74]:
# Qual foi o mês com mais vendas?
base.groupby([base['Data da Venda'].dt.month])[['Receita']].sum().sort_values(by=['Receita']).tail(1)

Receita
Data da Venda         
6              52340.0

In [79]:
# Qual foi a categoria mais lucrativa?
base.groupby(['Categoria'])[['Lucro']].sum().sort_values(by=["Lucro"]).tail(1)

Lucro
Categoria           
Informática  29530.0

In [52]:
# Qual cliente mais comprou?

In [81]:
# Qual estado gerou mais receita?
base.groupby(['Estado'])[['Receita']].sum().sort_values(by=["Receita"]).tail(1)

Receita
Estado         
RJ      60030.0

___

### 📌 Parte 3: Visualizações (Matplotlib e Seaborn)

- 📊 Gráfico de barras com Top 5 produtos mais vendidos
- 📈 Gráfico de linha com vendas mensais
- 📉 Gráfico de barras com lucro por categoria
- 🔥 Heatmap com correlação entre Receita, Custo, Lucro, Quantidade
- 🥧 Gráfico de pizza: proporção de vendas por categoria

___

### 📌 Parte 4: Exportação de Resultados

- Criar um resumo em DataFrame com:
- Receita total
- Lucro total
- Produto campeão de vendas
- Exportar esse resumo para um CSV (resumo_final.csv)
- Salvar os gráficos como PNGs

___

# 📈 Insights Comerciais

## 🔹 1. Produtos mais vendidos

- Quais produtos saem mais?
- Qual o ticket médio de cada produto?

📌 Exemplo de pergunta: _`"Devemos manter mais estoque de quais itens?"`_

## 🔹 2. Categorias com maior volume de vendas

- Qual categoria mais vendeu em quantidade e em valor?
- Existe alguma categoria com desempenho fraco?

📌 Exemplo:

`"A categoria ‘Celulares’ representa 40% do faturamento, mas apenas 20% das unidades vendidas. Isso indica produtos com ticket médio mais alto."`

## 🔹 3. Receita por estado (ou região)

- Quais estados compram mais?
- Há locais com baixo desempenho?

📌 Exemplo:

`"O estado de SP gerou 35% da receita total — será que vale investir em marketing local?"`

## 🔹 4. Lucro por produto e por categoria

- Quais produtos são mais lucrativos?
- Existe algum produto que vende muito, mas tem baixo lucro?

📌 Exemplo:

`"O Mouse Gamer vende muito, mas a margem de lucro é baixa — talvez seja hora de reajustar o preço."`

## 🔹 5. Desempenho mensal

- Qual mês teve maior receita?
- Há sazonalidade? (ex: aumento em novembro/dezembro)

📌 Exemplo:

`"As vendas disparam em dezembro. Devemos nos preparar com campanhas promocionais nesse período."`


## Análises financeiras

### 🔹 6. Lucro total e margem de lucro

- Qual a margem de lucro média da empresa?
- A margem está aumentando ao longo dos meses?

📌 Exemplo:

`"A margem média está em 20%, mas em janeiro foi 15% — houve algo errado naquele mês?"`

## 📊 Visualmente, com Matplotlib/Seaborn, você pode:

- Comparar lucro por categoria com gráficos de barras
- Ver a evolução das vendas com gráficos de linha
- Criar heatmaps de correlação para explorar relações entre preço, quantidade, receita e lucro
- Gerar gráficos de pizza para proporções de vendas